### ISSUE TITLE
MTOUCH : error MT2002: Failed to resolve assembly: 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'

### ISSUE THREAD

**Issue by kdubau** [xamarin/xamarin-macios#4237]
### Description

Uncovered this with regression protection running against master bits (11.13.0.9 currently). Of the apps in regression protection it looks like only this one is affected. The same app commit builds with15.7 bits (11.11).

### Steps to Reproduce

1. Install XI 11.13.0.9
2. Clone the SeeingAI app (internal)
3. Build it in Debug config

### Expected Behavior

Successful build

### Actual Behavior

```
2018-06-11T21:15:43.5986720Z (_CompileToNative target) -> 
2018-06-11T21:15:43.6019110Z   MTOUCH : error MT2002: Failed to resolve assembly: 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' [/Users/vsts/agent/2.134.2/work/1/s/iOS/App/SeeingAI.iOS.csproj]
2018-06-11T21:15:43.6036710Z 
2018-06-11T21:15:43.6053430Z     85 Warning(s)
2018-06-11T21:15:43.6070130Z     1 Error(s)
```

### Environment

```
Xcode | 9.4.0 
Xamarin.iOS | 11.13.0.9 
Xamarin.Mac | 4.5.0.382
Visual Studio for Mac | 7.6.0.1566
Mono | 5.14.0.121
```

### Build Logs

Failed Master (11.13.0.9): http://build.devdiv.io/1764931 (see Debug config)

Successful 15.7 RTW (11.10.1.177): http://build.devdiv.io/1745621 (see Debug config, Release is failed due to https://github.com/xamarin/xamarin-macios/issues/4235)

### Example Project (If Possible)

Contact me on Slack and we can work out getting you access to SeeingAI app.

**Comment by kdubau** [comment]
I'll update with 15.8 results as soon as I have them (in progress).

**Comment by spouliot** [comment]
The original error occurs earlier, there's more than one (two, maybe three) versions of the same assembly being used in the project. 

```
2018-06-11T21:15:14.5119040Z   No way to resolve conflict between "Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" and "Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed". Choosing "Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" arbitrarily.
```

also (3rd one ?)

```
2018-06-11T21:15:14.5590930Z   Consider app.config remapping of assembly "Newtonsoft.Json, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed" from Version "9.0.0.0" [] to Version "10.0.0.0" [/Users/vsts/.nuget/packages/newtonsoft.json/10.0.3/lib/netstandard1.3/Newtonsoft.Json.dll] to solve conflict and get rid of warning.
```

Like mentioned what `mtouch` gets is random - so your results will vary based on whatever ends up in the provided arguments.

**Comment by rolfbjarne** [comment]
@spouliot this is fact a regression in d15-8, it regressed here: https://github.com/mono/linker/commit/2dee7d0dddd7b793a8536a5a42d86541b0af3c46#diff-9a45f791ffe679ddd8307998bd695246R252

Previously the `AssemblyResolutionException` were swallowed, they're not anymore, bubbling up and eventually causing the mtouch error.

**Comment by kdubau** [comment]
This issue also exists in 15.8 (11.14.0.4) http://build.devdiv.io/1768483 (see Debug config build).

**Comment by spouliot** [comment]
Re-opning for possible 15.8 SR

**Comment by spouliot** [comment]
The fix is part of XI 12.0 (a 15.8 SR to support Xcode 10) and will be in XM 5.0 (15.9)
